Terminate bad nodes feature (#616)
giantswarm/docs: src/content/basics/automatic-termination-of-bad-nodes/index.md

---

> title: Automatic termination of bad nodes

Reviewer: ```suggestion
title: Automatic termination of unhealthy nodes
```
For alignment

description: A general description of a feature that can detect and terminate unhealthy nodes in the tenant cluster.
date: 2020-11-05
weight: 120
type: page
categories: ["basics"]
---

# Automatic termination of unhealthy nodes

## Overview

> Giant Swarm's `terminate-unhealthy-nodes` feature helps you keep the nodes in your cluster in a healthy, running state. When enabled, a service makes periodic checks on the health state of each node in your cluster. If a node fails consecutive health checks over an extended time period, it will be drained and terminated.

Reviewer 2: ```suggestion
Giant Swarm's `terminate-unhealthy-nodes` feature helps you keep the nodes in your cluster in a healthy, running state. When enabled, a service makes periodic checks on the health of each node in your cluster. If a node fails consecutive health checks over an extended time period, it will be drained and terminated.
```

> This is an `alpha` feature avaiable on theese providers:

Reviewer: ```suggestion
This is an `alpha` feature available on these providers:
```

> * AWS - from `12.6.0` release

Reviewer 2: ```suggestion
* AWS - `12.6.0` release onwards
```

## Technical details

> The node's health status is used to determine if a node needs to be terminated. A node reporting a Ready status is considered healthy. If a node reports consecutive unhealthy status reports for a given time threshold. An unhealthy status can mean:

Reviewer: ```suggestion
The node's health status is used to determine if a node needs to be terminated. A node reporting a `Ready` status is considered healthy. If a node reports an unhealthy status continuously for a given time threshold it will be recycled. 
```

> The node's health status is used to determine if a node needs to be terminated. A node reporting a Ready status is considered healthy. If a node reports consecutive unhealthy status reports for a given time threshold. An unhealthy status can mean:

Reviewer 2: ```suggestion
The node's health is used to determine if a node needs to be terminated. A node reporting a Ready status is considered healthy. If a node reports consecutive unhealthy status reports for a given time threshold. An unhealthy status can mean:
```

> * A node reports a NotReady status on consecutive checks over the given time threshold (approximately 15 minutes).

Reviewer: ```suggestion
An unhealthy status means the `kubelet` on a given node has reported `NotReady` status on consecutive checks over a certain time threshold (approximately 15 minutes).
```
I  remove the list because there was only single item and I put a single line

## Enabling terminate-unhealthy-nodes feature

> You can enable this feature on a cluster basis. To enable this feature you need to set annotation `alpha.node.giantswarm.io/terminate-unhealthy` on the `AWSCluster` CR. The value can be anything as only presence of that annotation is checked.

Reviewer: ```suggestion
You can enable this feature on a cluster basis. To enable this feature you need to set the annotation `alpha.node.giantswarm.io/terminate-unhealthy` on the `AWSCluster` custom resource. The value can be anything as the only presence of that annotation is checked.
```

> If you want to disable the feature, just remove the annotations from the `AWSCluster` CR.

Reviewer: ```suggestion
If you want to disable the feature you must remove the annotation from the `AWSCluster` custom resource.
```

The feature is disabled by default.

> ### example

Reviewer: ```suggestion
### Example
```
